ygnmi/ygnmi.go

```diff
@@ -326,6 +326,9 @@ func Watch[T any](ctx context.Context, c *Client, q SingletonQuery[T], pred func
 		gs := q.goStruct()
 		for {
 			select {
+			case <-ctx.Done():
+				w.errCh <- nil
```

Comment:
this should send ctx.Err() to the errChan, so the caller can check if the context was cancelled or the predicate returned successfully

Reply:
Good catch, done.